odoo/addons/base/tests/test_ir_actions.py

```diff
@@ -198,6 +199,30 @@ def test_30_crud_write(self):
         self.assertEqual(len(partner), 1, 'ir_actions_server: TODO')
         self.assertEqual(partner.city, 'OrigCity', 'ir_actions_server: TODO')
 
+    def test_35_crud_write_selection(self):
+        # Don't want to use res.partner because no 'normal selection field' exists there
+        # we'll use a speficic action for this test instead of the one from the test setup
+        # Do: update country name_position field
+        selection_value = self.res_country_name_position_field.selection_ids.filtered(lambda s: s.value == 'after')
+        action = self.env['ir.actions.server'].create({
+            'name': 'TestAction',
+            'model_id': self.res_country_model.id,
+            'model_name': 'res.country',
+            'state': 'object_write',
+            'update_field_id': self.res_country_name_position_field.id,
+            'selection_value': selection_value.id,
+        })
+        action._set_selection_value()  # manual onchange
```

Comment:
assert before onchange that action.value is falsy

Reply:
this wouldn't change anything in the behaviour of the onchange method so not sure what it would test?

Reply:
OK then, no need to change anything.

It's just that I usually like to see in a test:
```
# setup
...

# scenario A
assertBefore
effect
assertAfter

# other scenario
...
```